addons/mail/models/discuss/res_users.py

```diff
@@ -62,6 +62,11 @@ def _init_store_data(self, store: Store):
         store.add_global_values(
             hasGifPickerFeature=bool(get_param("discuss.tenor_api_key")),
             hasMessageTranslationFeature=bool(get_param("mail.google_translate_api_key")),
+            hasCannedResponses=bool(self.env["mail.canned.response"].sudo().search([
+                "|",
+                ("create_uid", "=", self.env.user.id),
+                ("group_ids", "in", self.env.user.groups_id.ids),
+            ], limit=1)) if self.env.user else False,
```

Comment:
You might as well directly send the canned response to the user instead.

Reply:
You can have a lot of canned responses, this is fetched while you make the search.

That's not a good idea to give all canned response at page load, we have already all pinned channels and this introduce performance issues that eventually need to be solved.